### PR TITLE
Fix error when safe yaml is only partially loaded

### LINKED ISSUE #2218: undefined method `map' for nil:NilClass
Hello.  In running rubocop today, I received an error message where my code had previously passed a few days before (presumably with the .33 version).  I admit I am relatively new to the Ruby eco-system and do not have enough knowledge to be able to troubleshoot so I figured I would submit an issue.  

Thank you ever so much and thanks for a wonderful product!

```
$ bundle exec rake style
Running RuboCop...
undefined method `map' for nil:NilClass
/home/travis/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/psych.rb:287:in `safe_load'
/home/travis/.rvm/gems/ruby-2.2.2/gems/rubocop-0.34.0/lib/rubocop/config_loader.rb:151:in `yaml_safe_load'
/home/travis/.rvm/gems/ruby-2.2.2/gems/rubocop-0.34.0/lib/rubocop/config_loader.rb:138:in `load_yaml_configuration'
/home/travis/.rvm/gems/ruby-2.2.2/gems/rubocop-0.34.0/lib/rubocop/config_loader.rb:27:in `load_file'
/home/travis/.rvm/gems/ruby-2.2.2/gems/rubocop-0.34.0/lib/rubocop/config_loader.rb:85:in `configuration_from_file'
/home/travis/.rvm/gems/ruby-2.2.2/gems/rubocop-0.34.0/lib/rubocop/config_store.rb:40:in `for'
/home/travis/.rvm/gems/ruby-2.2.2/gems/rubocop-0.34.0/lib/rubocop/target_finder.rb:112:in `excluded_dirs'
/home/travis/.rvm/gems/ruby-2.2.2/gems/rubocop-0.34.0/lib/rubocop/target_finder.rb:90:in `find_files'
/home/travis/.rvm/gems/ruby-2.2.2/gems/rubocop-0.34.0/lib/rubocop/target_finder.rb:60:in `target_files_in_dir'
/home/travis/.rvm/gems/ruby-2.2.2/gems/rubocop-0.34.0/lib/rubocop/target_finder.rb:31:in `find'
/home/travis/.rvm/gems/ruby-2.2.2/gems/rubocop-0.34.0/lib/rubocop/runner.rb:45:in `find_target_files'
/home/travis/.rvm/gems/ruby-2.2.2/gems/rubocop-0.34.0/lib/rubocop/runner.rb:29:in `run'
/home/travis/.rvm/gems/ruby-2.2.2/gems/rubocop-0.34.0/lib/rubocop/cli.rb:26:in `run'
/home/travis/.rvm/gems/ruby-2.2.2/gems/rubocop-0.34.0/lib/rubocop/rake_task.rb:50:in `run_cli'
/home/travis/.rvm/gems/ruby-2.2.2/gems/rubocop-0.34.0/lib/rubocop/rake_task.rb:38:in `run_main_task'
/home/travis/.rvm/gems/ruby-2.2.2/gems/rubocop-0.34.0/lib/rubocop/rake_task.rb:30:in `block (2 levels) in initialize'
/home/travis/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/rake/file_utils_ext.rb:58:in `verbose'
/home/travis/.rvm/gems/ruby-2.2.2/gems/rubocop-0.34.0/lib/rubocop/rake_task.rb:26:in `block in initialize'
/home/travis/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/rake/task.rb:240:in `call'
/home/travis/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/rake/task.rb:240:in `block in execute'
/home/travis/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/rake/task.rb:235:in `each'
/home/travis/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/rake/task.rb:235:in `execute'
/home/travis/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/rake/task.rb:179:in `block in invoke_with_call_chain'
/home/travis/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/monitor.rb:211:in `mon_synchronize'
/home/travis/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/rake/task.rb:172:in `invoke_with_call_chain'
/home/travis/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/rake/task.rb:201:in `block in invoke_prerequisites'
/home/travis/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/rake/task.rb:199:in `each'
/home/travis/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/rake/task.rb:199:in `invoke_prerequisites'
/home/travis/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/rake/task.rb:178:in `block in invoke_with_call_chain'
/home/travis/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/monitor.rb:211:in `mon_synchronize'
/home/travis/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/rake/task.rb:172:in `invoke_with_call_chain'
/home/travis/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/rake/task.rb:165:in `invoke'
/home/travis/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/rake/application.rb:150:in `invoke_task'
/home/travis/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/rake/application.rb:106:in `block (2 levels) in top_level'
/home/travis/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/rake/application.rb:106:in `each'
/home/travis/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/rake/application.rb:106:in `block in top_level'
/home/travis/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/rake/application.rb:115:in `run_with_threads'
/home/travis/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/rake/application.rb:100:in `top_level'
/home/travis/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/rake/application.rb:78:in `block in run'
/home/travis/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/rake/application.rb:176:in `standard_exception_handling'
/home/travis/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/rake/application.rb:75:in `run'
/home/travis/.rvm/rubies/ruby-2.2.2/lib/ruby/gems/2.2.0/gems/rake-10.4.2/bin/rake:33:in `<top (required)>'
/home/travis/.rvm/gems/ruby-2.2.2/bin/rake:23:in `load'
/home/travis/.rvm/gems/ruby-2.2.2/bin/rake:23:in `<main>'
/home/travis/.rvm/gems/ruby-2.2.2/bin/ruby_executable_hooks:15:in `eval'
/home/travis/.rvm/gems/ruby-2.2.2/bin/ruby_executable_hooks:15:in `<main>'
RuboCop failed!
```

### PR DESCRIPTION
**Pull request by maxjacobson**
This should fix #2218 and #2215

The error was introduced here: https://github.com/bbatsov/rubocop/commit/2c32a61f5bf144a3c824c4287398b317a0f51096 and kind of here: https://github.com/sferik/rails_admin/commit/c1a495edf40124dbbb391acb4e9240f2ac15f8a9 and can be resolved by following the example of that second commit

### CODE REVIEW

**Comment by maxjacobson** [comment]
Tried three approaches (https://github.com/maxjacobson/rubocop/commit/d22f633d92a3c2d5663f8c130d13fabda8743000, https://github.com/maxjacobson/rubocop/commit/12849814f33b746e52d7eb2a82278aa7fb389784, https://github.com/maxjacobson/rubocop/commit/8b9a2b39913044623012636fa3714cbc72bb3a05) and I think the last one is best, so I'm squashing.

File: spec/rubocop/config_loader_spec.rb [hunk 1]
```diff
@@ -292,6 +292,8 @@
       expect(configuration).to eq({})
     end
 
+    # TODO: figure out why these specs print a green dot even when they fail
+    # (probably related to the process shenanigans)
```
Comment (maxjacobson) [inline spec/rubocop/config_loader_spec.rb]: this is a bit concerning, I think, but I have NO idea how to resolve it :grimacing:

**Comment by bbatsov** [comment]
//cc @eitoball

Those safe_yml issues are killing me...

**Comment by maxjacobson** [comment]
The error kind of makes sense in retrospect / after reading the safe yaml readme. They encourage you to choose whether you want to monkey patch YAML or not, so rubocop can't assume that the monkey patch occurred

**Comment by maxjacobson** [comment]
Pushed an update to avoid duplicating too much code, and to assert on the successful execution of the subprocess (otherwise failing tests are green)

File: spec/rubocop/config_loader_spec.rb [hunk 2]
```diff
@@ -300,25 +300,27 @@
...
-          pid = ::Process.fork do
-            # Need to write coverage result under different name
-            if defined?(SimpleCov)
-              SimpleCov.command_name "rspec_#{Process.pid}"
-              SimpleCov.pid = Process.pid
-            end
-
+      context 'when it is fully required' do
+        it 'de-serializes Regexp class' do
+          in_its_own_process do
```
Comment (rud) [inline spec/rubocop/config_loader_spec.rb]: That's a pretty neat helper for clearing up the intention.

Comment (maxjacobson) [inline spec/rubocop/config_loader_spec.rb]: I'm about to update it to, I think, make it even more intentional :whoa:

**Comment by maxjacobson** [comment]
Squashed again, but if you're interested in the changes since the last commit, they were: https://github.com/maxjacobson/rubocop/commit/eebc66309bcc5756313f89519aa274330d0ba2e9 and https://github.com/maxjacobson/rubocop/commit/78d65139605d4ff3953606d34c589071ce23bbd4

**Comment by maxjacobson** [comment]
One more :blush: https://github.com/maxjacobson/rubocop/commit/5d6fe78ea3a9c79b151fe8002f6015dc9b593f32

**Comment by bbatsov** [comment]
:+1:

**Pull request merged**